packages/compile-solidity/compilerSupplier/index.js

```diff
@@ -8,9 +8,8 @@ const defaultSolcVersion = "0.5.16";
 
 class CompilerSupplier {
   constructor({ events, solcConfig }) {
-    const { version, docker, compilerRoots, parser, spawn } = solcConfig;
+    const { version, docker, compilerRoots, spawn } = solcConfig;
     this.events = events;
-    this.parser = parser;
     this.version = version ? version : defaultSolcVersion;
```

Comment:
Hm, it's a bit tricky here, because the default is necessary for the compiler supplier's interface, and I don't want to introduce the @truffle/config dependency. Note how this doesn't actually take a config (it takes that `{ solcConfig, ... }` options object)

Reply:
Note I have no idea if changing this is actually a good idea; I think leaving it alone is fine!  I don't see how this would introduce a dependency though?  Like how would putting a default version in the config defaults introduce a dependency here?

Anyway yeah probably makes sense to leave this alone for now, or at least to address it in a separate PR.

Reply:
@eggplantzzz, I don't understand the question.